src/cli/commands/ping.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const print = require('../utils').print
+
+module.exports = {
+  command: 'ping <peerId>',
+
+  describe: 'Measure the latency of a connection',
+
+  builder: {
+    count: {
+      alias: 'n',
+      type: 'integer',
+      default: 10
+    }
+  },
+
+  handler (argv) {
+    const peerId = argv.peerId
+    const count = argv.count || 10
+
+    print('PING ' + peerId)
+
+    let noOfTimes = 0
+    let totalTime = 0
+
+    const pingCb = (err, p) => {
+      if (err) {
+        throw err
+      }
+      let time = p.Time
+      totalTime = totalTime + time
+      noOfTimes = noOfTimes + 1
+      print('Pong received: time=' + time + ' ms')
+      if (noOfTimes === count) {
+        print('Average latency: ' + totalTime / count + 'ms')
+      }
+    }
+
+    for (let i = 0; i < count; i++) {
+      argv.ipfs.ping(peerId, pingCb)
+    }
```

Comment:
A ping with count 10 is a ping with 10 packets, not 10 ping calls. You need to count the number of pings that come in p.

Reply:
I was able to use the ping object and stream the times to the api.
But I have run into a problem. In [js-ipfs-api](https://github.com/ipfs/js-ipfs-api/blob/master/src/ping.js#L27) the response assumes there will be one ping request. 

So even if I send I stream multiple ping responses only the first one is returned.

My understanding of the flow when someone types jsipfs ping <peerId> is
1.  cli/commands/ping.js -> this calls the ipfs-api ping
2. node_modules/ipfs-api/src/ping.js -> this calls the http api ping
3. http/api/resources/ping.js -> this calls the component ping
4. core/components/ping.js which would then return the ping object
5. http/api/resources/ping.js would then stream the times to the ipfs-api and end the stream when number of pings are complete.

The problem is [js-ipfs-api](https://github.com/ipfs/js-ipfs-api/blob/master/src/ping.js#L27) is expecting only one response ping object. Is there a workaround I'm missing or should I create an issue inn js-ipfs-api ping and fix that. I just want to confirm its an issue. :)


Also, irrespective of the api I was able to get the curl outputs for both the go-ipfs http api and js-ipfs http api to be the same . I have attached images for the same. But again I find one inconsistency . Since we are using new Date() in [js-libp2p-ping](https://github.com/libp2p/js-libp2p-ping/blob/master/src/ping.js#L42) we are not getting the same unit as go-ipfs. go-ipfs returns in nanosecond and js-ipfs returns the time in milliseconds.  Let me know whether I should create a issue and fix that. I would be glad to do it. I just want to confirm its an issue. :)

Curl response to js-ipfs
<img width="1012" alt="screen shot 2018-02-17 at 6 37 38 pm" src="https://user-images.githubusercontent.com/5201724/36347754-3938f77a-1414-11e8-99db-fd6ed4e828a7.png">

Curl response to go-ipfs
<img width="927" alt="screen shot 2018-02-17 at 6 36 28 pm" src="https://user-images.githubusercontent.com/5201724/36347755-3dcfbf76-1414-11e8-8971-26b1a1e18cf6.png">